unit-tests/live/image-quality/test-image-quality.py

```diff
@@ -0,0 +1,148 @@
+        k = cv2.waitKey(33)
+        if k == 27:  # Esc key to stop
+            cv2.destroyAllWindows()
+            break
+        elif k == -1:  # normally -1 returned,so don't print it
+            pass
+
+
+def save_image(depth, color, filename, showImage):
+    colorizer = rs.colorizer()
+    depth_image = np.asanyarray(colorizer.colorize(depth).get_data())
+    color_image = np.asanyarray(color.get_data())
+
+    depth_rows, _, _ = depth_image.shape
+    color_rows, _, _ = color_image.shape
+
+    # resize the image with the higher resolution to look like the smaller one
+    if depth_rows < color_rows:
+        color_image = zoom(color_image, (depth_rows/color_rows, depth_rows/color_rows, 1))
+    elif color_rows < depth_rows:
+        depth_image = zoom(depth_image, (color_rows/depth_rows, color_rows/depth_rows, 1))
+
+    img = np.concatenate((depth_image, color_image), axis=1)
+
+    if filename != "":
+        cv2.imwrite(filename, img)
+        cv2.imwrite("test.png",depth_image)
+        cv2.imwrite("test2.png",color_image)
+
+    if showImage:
+        display_image("Depth Stream", img)
+
+
+def setup(config, laserOn):
+    pipeline = rs.pipeline()
+    pipeline_profile = pipeline.start(config)
+
+    if laserOn is False:
+        sensor = pipeline_profile.get_device().first_depth_sensor()
+        sensor.set_option(rs.option.laser_power, 0)
+        sensor.set_option(rs.option.emitter_enabled, 0)
+
+    frames = pipeline.wait_for_frames()
+    for i in range(30):
+        frames = pipeline.wait_for_frames()
+    depth = frames.get_depth_frame()
+    color = frames.get_color_frame()
+    return pipeline, depth, color
+
+
+def get_distances(depth):
+    MAX_METERS = 10  # max distance that can be detected, in meters
+    dists = {}
+    total = 0
+    for y in range(480):
+        for x in range(640):
+            dist = depth.get_distance(x, y)
+            if dist >= MAX_METERS:  # out of bounds, assuming it is a junk value
+                continue
+            # we only distinguish between two pixels if they are different more than 10cm apart
+            # we can get a more accurate measure if we round it less, say round(dist,2)*100 to get 1cm difference noted
+            dist = int(round(dist, 1) * 10)
+            if dists.get(dist) is not None:
+                dists[dist] += 1
+            else:
+                dists[dist] = 1
+            total += 1
+    # print(dists)
+    return dists, total
+
+
+def check_depth(config, laserOn=True, filename="", showImage: bool = False):
+    """
+    Checks if the camera is showing a frame with a meaningful depth
+    the higher the detail level is, the more it is sensitive to distance
+    eg: 1 for a difference of 1 meter, 100 for a diff of 1cm
+
+    returns true iff is it does
+    """
+
+    pipeline, depth, color = setup(config, laserOn)
+
+    if not depth or not color:
+        print("Error getting depth / color frames")
```

Comment:
Seperate to two `if` statements to supply more accurate error descriptions.
Also, if there is no color why do we fail the depth check?